terminal/glint.go

```diff
@@ -34,15 +40,48 @@ func (ui *glintUI) Close() error {
 }
 
 func (ui *glintUI) Input(input *Input) (string, error) {
-	return "", ErrNonInteractive
+	var buf bytes.Buffer
+
+	// Write the prompt, add a space.
+	ui.Output(input.Prompt, WithStyle(input.Style), WithWriter(&buf))
+	fmt.Fprint(color.Output, strings.TrimRight(buf.String(), "\r\n"))
+	fmt.Fprint(color.Output, " ")
+
+	// Ask for input in a go-routine so that we can ignore it.
+	errCh := make(chan error, 1)
+	lineCh := make(chan string, 1)
+	go func() {
+		var line string
+		var err error
+		if input.Secret && isatty.IsTerminal(os.Stdin.Fd()) {
+			line, err = speakeasy.Ask("")
+		} else {
+			r := bufio.NewReader(os.Stdin)
+			line, err = r.ReadString('\n')
+		}
+		if err != nil {
+			errCh <- err
+			return
+		}
+
+		lineCh <- strings.TrimRight(line, "\r\n")
+	}()
+
+	select {
+	case err := <-errCh:
+		return "", err
+	case line := <-lineCh:
+		return line, nil
+	case <-ui.ctx.Done():
+		// Print newline so that any further output starts properly
+		fmt.Fprintln(color.Output)
+		return "", ui.ctx.Err()
+	}
 }
 
 // Interactive implements UI
 func (ui *glintUI) Interactive() bool {
-	// TODO(mitchellh): We can make this interactive later but Glint itself
-	// doesn't support input yet. We can pause the document, do some input,
-	// then resume potentially.
-	return false
+	return isatty.IsTerminal(os.Stdin.Fd())
```

Comment:
If glint still doesn't support interactive, maybe you could submit an upstream PR. Worth it?